javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -482,18 +482,17 @@
     }
 
     /**
-     * Transposes the rows and columns of a {@link Vector}.
+     * Transposes the rows and columns of a {@link Vector} matrix.
      *
      * @param rows to be transposed.
      * @return a transposed {@link Vector}.
```

Comment:
I like to have `matrix` because it tells us something about the domain.

@paplorinc Please add `@throws IllegalArgumentException if rows is not a regular n x m matrix.` (or similar - your call) to the public `transpose` methods.